docs: add C4 architecture diagrams to enhance documentation (#4867)
equinor/design-system: README.md

  eds-tailwind/          # Tailwind CSS plugin
```
  
## Architecture

### System Context

> EDS serves three personas and delivers through four channels: Figma assets for designers, npm packages for developers, a documentation hub for both, and a CDN that provides fonts and CSS to all Equinor web applications utlilizing EDS. The system is built as a monorepo with CI/CD pipelines that automate testing, building and publishing to npm and the CDN.

Reviewer: ```suggestion
EDS serves three personas and delivers through four channels: Figma assets for designers, npm packages for developers, a documentation hub for both, and a CDN that provides fonts and CSS to all Equinor web applications utilizing EDS. The system is built as a monorepo with CI/CD pipelines that automate testing, building and publishing to npm and the CDN.
```

Reviewer: or just 'using'

```mermaid
C4Context
  title Equinor Design System – System Context

  Person(designer, "Designer", "Creates and maintains UI designs following the Equinor design language")
  Person(developer, "Developer", "Builds web applications using EDS components and tokens")
  
  System_Boundary(eds, "Equinor Design System") {
    System_Ext(figma, "Figma", "Source of truth for design tokens, components and assets")
    System_Ext(npm, "npm Registry", "Distributes EDS packages: eds-core-react, eds-tokens, eds-icons, etc.")
    System(eds, "Equinor Design System<br>equinor/design-system | eds.equinor.com", "Documentation hub, GitHub monorepo and CI/CD pipelines delivering reusable UI components, design tokens, icons and typography")
  }
  

>   System_Boundary(applications, "Equinor Web Applications", "All internal web applications built by and for Equinor") {

Reviewer: The "Equinor Web Applications" boundary contains only End User and CDN — but no actual application node. As a result, line 68:

```
Rel(enduser, cdn, "Loads fonts and CSS via applications", "HTTPS")
```

skips a layer: the label says "via applications" but there's no application node. In reality it's `App → CDN` and `EndUser → App`. The Developer persona also has no link to the apps they build.

Also note that End User is currently inside the apps boundary — in C4, Person nodes should sit outside system boundaries since they interact with systems, not constitute them.

Suggest moving End User out of the boundary, adding an `app` system inside, and rewiring the relations. Replace lines 54–57:

```
Person(enduser, "End User", "Uses Equinor web applications built with EDS")

System_Boundary(applications, "Equinor Web Applications", "All internal web applications built by and for Equinor") {
  System(app, "Equinor web app", "Internal application built with EDS")
  System_Ext(cdn, "CDN", "Serves EDS fonts, icons and CSS to all Equinor web applications across the tenant")
}
```

And replace line 68 with the through-app routing:

```
Rel(developer, app, "Builds")
Rel(enduser, app, "Uses")
Rel(app, npm, "Depends from")
Rel(app, cdn, "Loads fonts and CSS from", "HTTPS")
```

With the new developer → app and app → npm relations, the direct developer → npm line is redundant — the install is part of "Builds". Suggest dropping it for clarity.

Author: I understand, but I have to experiment some to make this look clean.

Author: What do you think about something like this?

```mermaid
C4Context
  title Equinor Design System - System Context
  
  Person(enduser, "End User", "Uses Equinor web applications built with EDS")

  Boundary(applications, "Equinor Web Applications", "All internal web applications built by and for Equinor") {
    System(app, "Equinor web app", "Internal application built with EDS")
    System_Ext(cdn, "CDN", "Serves EDS fonts, icons and CSS to all Equinor web applications across the tenant")
  }
  
  Boundary(personas, "", "") {
    Person(designer, "Designer", "Creates and maintains UI designs following the Equinor design language")
    Person(developer, "Developer", "Builds web applications using EDS components and tokens")
  }

  System_Boundary(eds, "Equinor Design System") {
    System_Ext(npm, "npm Registry", "Distributes EDS packages: eds-core-react, eds-tokens, eds-icons, etc.")
    System_Ext(figma, "Figma", "Source of truth for design tokens, components and assets")
    System(eds, "Equinor Design System<br>equinor/design-system | eds.equinor.com", "Documentation hub, GitHub monorepo and CI/CD pipelines delivering reusable UI components, design tokens, icons and typography")
  }

  Rel(designer, eds, "Uses design assets<br>and documentation")
  UpdateRelStyle(designer, eds, $offsetX="-340", $offsetY="-50")
  Rel(developer, eds, "Uses documentation<br>and developer tools")
  UpdateRelStyle(developer, eds, $offsetX="-90", $offsetY="-30")

  BiRel(eds, figma, "Syncs assets<br>and tokens with")
  UpdateRelStyle(eds, figma, $offsetX="-80", $offsetY="-50")
  Rel(eds, npm, "Publishes<br>packages to")
  UpdateRelStyle(eds, npm, $offsetX="-270", $offsetY="30")
  Rel(eds, cdn, "Publishes fonts and<br>static assets to")
  UpdateRelStyle(eds, cdn, $offsetX="80", $offsetY="-50")

  Rel(developer, app, "Builds")
  Rel(enduser, app, "Uses")
  Rel(app, npm, "Depends from")
  UpdateRelStyle(app, npm, $offsetX="-40", $offsetY="250")
  Rel(app, cdn, "Loads fonts<br>and CSS from", "HTTPS")
  UpdateRelStyle(app, cdn, $offsetX="-60", $offsetY="-30")
  Rel(designer, figma, "Uses EDS components in", "Figma")
  UpdateRelStyle(designer, figma, $offsetX="-200", $offsetY="-10")

  UpdateLayoutConfig($c4ShapeInRow="4", $c4BoundaryInRow="1")
```

Author: Alternative 2:

@pomfrida 
Instead of app depending on libraries, could we say that the developer builds with packages from npm? otherwise it could appear as if the app is dependent on npm at runtime.

```mermaid
C4Context
  title Equinor Design System - System Context
  
  Boundary(applications, "Equinor Web Applications", "All internal web applications built by and for Equinor") {
    System(app, "Equinor web app", "Internal application built with EDS")
    System_Ext(cdn, "CDN", "Serves EDS fonts, icons and CSS to all Equinor web applications across the tenant")
  }

  Boundary(personas, "", "") {
    Person(enduser, "End User", "Uses Equinor web applications built with EDS")
    Person(designer, "Designer", "Creates and maintains UI designs following the Equinor design language")
    Person(developer, "Developer", "Builds web applications using EDS components and tokens")
  }

  System_Boundary(eds, "Equinor Design System") {
    System(eds, "Equinor Design System<br>equinor/design-system | eds.equinor.com", "Documentation hub, GitHub monorepo and CI/CD pipelines delivering reusable UI components, design tokens, icons and typography")
    System_Ext(figma, "Figma", "Source of truth for design tokens, components and assets")
    System_Ext(npm, "npm Registry", "Distributes EDS packages: eds-core-react, eds-tokens, eds-icons, etc.")
  }

  Rel(designer, eds, "Uses design assets<br>and documentation")
  UpdateRelStyle(designer, eds, $offsetX="-20", $offsetY="-50")
  Rel(developer, eds, "Uses documentation<br>and developer tools")
  UpdateRelStyle(developer, eds, $offsetX="-70", $offsetY="-40")

  BiRel(eds, figma, "Syncs assets<br>and tokens with")
  UpdateRelStyle(eds, figma, $offsetX="-80", $offsetY="-10")
  Rel(eds, npm, "Publishes<br>packages to")
  UpdateRelStyle(eds, npm, $offsetX="-10", $offsetY="-20")
  Rel(eds, cdn, "Publishes fonts and<br>static assets to")
  UpdateRelStyle(eds, cdn, $offsetX="70", $offsetY="-340")

  Rel(developer, app, "Builds & deploys")
  UpdateRelStyle(developer, app, $offsetX="200", $offsetY="90")
  Rel(enduser, app, "Uses")
  UpdateRelStyle(enduser, app, $offsetX="20", $offsetY="50")
  Rel(developer, npm, "Builds with<br>packages from", "npm")
  UpdateRelStyle(developer, npm, $offsetX="30", $offsetY="-140")
  Rel(app, cdn, "Loads fonts<br>and CSS from", "HTTPS")
  UpdateRelStyle(app, cdn, $offsetX="-60", $offsetY="-30")
  Rel(designer, figma, "Uses EDS components in", "Figma")
  UpdateRelStyle(designer, figma, $offsetX="-130", $offsetY="-150")

  UpdateLayoutConfig($c4ShapeInRow="3", $c4BoundaryInRow="1")
```

Reviewer: Thanks for the experiments — and you were actually right about the npm relation in your Alt 2. I initially thought we could fix the runtime ambiguity with wording alone (e.g. `"Bundled with packages from"` + `"build-time"` tag), but after some more digging it's clearer that on L1 all arrows should represent direct system interactions, and the build-time bundle is indirect. Moving it to `developer → npm` (your Alt 2 approach) is the right call.

I also found problem in the original structure: Figma and npm are marked as `System_Ext` but were placed *inside* the `System_Boundary(eds, ...)`, which is contradictory — external systems shouldn't sit inside the focused system's boundary. A strict L1 Context diagram should have one system in focus (EDS) with everything else flat around it, no nested boundaries.

Here's a version that keeps the original topology of the diagram but flat:
```mermaid
C4Context
  title System Context diagram for Equinor Design System

  Person(designer, "Designer", "Creates and maintains UI designs following the Equinor design language")
  Person(developer, "Developer", "Builds web applications using EDS components and tokens")
  Person(enduser, "End User", "Uses Equinor web applications built with EDS")

  System_Ext(figma, "Figma", "Source of truth for design tokens, components and assets")
  System(eds, "Equinor Design System", "Documentation hub, GitHub monorepo and CI/CD pipelines delivering reusable UI components, design tokens, icons and typography")
  System_Ext(app, "Equinor Web Applications", "Internal web apps built with EDS")

  System_Ext(cdn, "CDN", "Serves font files, icons and logo for Equinor web applications")
  System_Ext(npm, "npm Registry", "Distributes EDS packages: eds-core-react, eds-tokens, eds-icons, etc.")

  Rel(designer, eds, "Uses design assets and documentation")
  Rel(developer, eds, "Uses documentation and developer tools")
  Rel(designer, figma, "Uses EDS components in", "Figma")

  BiRel(eds, figma, "Syncs assets and tokens with")
  Rel(eds, npm, "Publishes packages to")
  Rel(eds, cdn, "Publishes fonts and static assets to")

  Rel(developer, npm, "Installs EDS packages from")
  Rel(developer, app, "Builds")
  Rel(enduser, app, "Uses")
  Rel(app, cdn, "Loads fonts from", "HTTPS")

  UpdateRelStyle(app, cdn, $offsetX="250", $offsetY="0")
  UpdateRelStyle(eds, figma, $offsetX="0", $offsetY="30")
  UpdateRelStyle(developer, npm, $offsetX="0", $offsetY="-60")
  UpdateRelStyle(designer, eds, $offsetX="-150", $offsetY="0")
  UpdateRelStyle(developer, eds, $offsetX="-80", $offsetY="-30")
  UpdateRelStyle(eds, cdn, $offsetX="-100", $offsetY="0")

  UpdateLayoutConfig($c4ShapeInRow="3", $c4BoundaryInRow="2")
```

A few notes on the choices:

- **Strict L1, no nested boundaries** — EDS is the single focused system; Figma, npm, CDN and Equinor Web Applications are flat external systems around it. Avoids the "external systems inside a system boundary" contradiction.
- **`Rel(developer, npm, "Installs EDS packages from")`** — direct system interaction (developer runs `npm install`), instead of the indirect `app → npm` build-time relation which doesn't really belong on an L1.
- **`Rel(app, cdn, "Loads fonts from", "HTTPS")`** kept — this *is* a direct runtime interaction (apps fetch fonts via HTTPS in production).
- **`UpdateRelStyle` offsets** — needed in a few places to clear label overlaps that Mermaid's auto-layout struggles with. 

If we want to highlight what's inside Equinor vs third-party, we could wrap the internal ones in an `Enterprise_Boundary` but that's something we could add later if we find necessary

Author: Awesome @pomfrida, looks a lot more compact and easy on the eyes! 

Updated with e4f7de6ddc0a8762aa8bb7bef72af5f24c40c920, one small difference is that i put cdn on the right side of npm, untangles some of the relations.

Reviewer: Thanks for letting me collab on this - I know I'm a little nit picky but I just learned this in school so it was fun to use it in practise 🙌  Great team work 👏

    Person(enduser, "End User", "Uses Equinor web applications built with EDS")

>     System_Ext(cdn, "CDN", "Serves EDS fonts, icons and CSS to all Equinor web applications across the tenant")

Reviewer: Only CSS is the font-face declarations, so saying CSS might be confusing as the main CSS is on npm. And the CDN is available to anyone with the link so 'across the tenant' phrasing isn't accurate

  ```suggestion
      System_Ext(cdn, "CDN", "Serves font files, icons and logo for Equinor web applications")
  ```

  }

  Rel(designer, eds, "Uses design assets and documentation")
  Rel(developer, eds, "Uses documentation and developer tools")

>   Rel(eds, figma, "Publishes design assets to")

Reviewer: The description calls Figma the "Source of truth for design tokens, components and assets", but the relation says:

```
Rel(eds, figma, "Publishes design assets to")
```

In practice the primary flow is the other way: `figma-broker` pulls icons from Figma into the repo, and `eds-tokens-sync` pulls tokens via `sync-figma-to-tokens`. Tokens additionally sync back via `sync-tokens-to-figma`  so it's two-way for tokens but Figma → code for icons and component specs.

Suggest using `BiRel` to express the bidirectional nature in one line, replacing line 62 with:

```
BiRel(eds, figma, "Syncs assets and tokens with")
```

(Verified locally: `BiRel` renders as a single line with arrow heads on both ends, much cleaner than two opposite `Rel` calls which overlap visually.)
